File: src/main/java/org/gecko/view/inspector/element/container/InspectorRegionColorItem.java
package org.gecko.view.inspector.element.container;

import javafx.scene.layout.HBox;
import org.gecko.view.inspector.InspectorColorPicker;
import org.gecko.view.inspector.element.InspectorElement;
import org.gecko.view.inspector.element.label.InspectorLabel;
import org.gecko.viewmodel.RegionViewModel;

public class InspectorRegionColorItem extends HBox implements InspectorElement<HBox> {

    public InspectorRegionColorItem(RegionViewModel regionViewModel) {
        getChildren()
                .addAll(
                        new InspectorLabel("L: Color"),
                        new InspectorColorPicker(regionViewModel.getColorProperty()));
    }

    @Override
    public HBox getControl() {
        return this;
    }
}

File: src/main/kotlin/org/gecko/view/inspector/element/container/InspectorRegionColorItem.kt
package org.gecko.view.inspector.element.container

import org.gecko.actions.ActionManager
import org.gecko.view.ResourceHandler
import org.gecko.view.inspector.element.InspectorColorPicker
import org.gecko.view.inspector.element.label.InspectorLabel
import org.gecko.viewmodel.Region

/**
 * Represents a type of [LabeledInspectorElement]. Contains an [InspectorLabel] and an
 * [InspectorColorPicker].
 */
class InspectorRegionColorItem(actionManager: ActionManager, Region: Region) :
    LabeledInspectorElement(
        InspectorLabel(ResourceHandler.Companion.color),
        InspectorColorPicker(actionManager, Region)
    )
